src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,48 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        int supplySum = 0;
+        int buySum = 0;
+
+        // Чтение данных из файла
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                String[] parts = line.split(",");
+                String operationType = parts[0];
+                int amount = Integer.parseInt(parts[1]);
+
+                // Увеличение суммы в зависимости от типа операции
+                if (operationType.equals("supply")) {
+                    supplySum += amount;
+                } else if (operationType.equals("buy")) {
+                    buySum += amount;
+                }
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't write data to file");
+        }
+
+        // Подсчет результата
+        int result = supplySum - buySum;
+
+        // Формирование строки отчета
+        String report = "supply," + supplySum + System.lineSeparator()
```

Comment:
Using String concatenation in a loop or similar structure can lead to performance issues due to the immutability of String objects. Consider using a `StringBuilder` to build the `report` string.